File: Selenium/Activities/Activity4.java
package examples;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.Color;

public class Activity4 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();

        driver.get("https://training-support.net/webelements/target-practice");

        System.out.println("Page title: " + driver.getTitle());

        String thirdHeader = driver.findElement(
                By.xpath("//h3[contains(text(),'#3')]")
        ).getText();
        System.out.println("Third header: " + thirdHeader);

        String colorValue = driver.findElement(
                By.xpath("//h5[contains(text(),'#5')]")
        ).getCssValue("color");

        Color color = Color.fromString(colorValue);
        System.out.println("RGB: " + color.asRgb());
        System.out.println("HEX: " + color.asHex());

        String purpleClasses = driver.findElement(
                By.xpath("//button[text()='Purple']")
        ).getDomAttribute("class");
        System.out.println("Purple classes: " + purpleClasses);

        String slateText = driver.findElement(
                By.xpath("//button[text()='Slate']")
        ).getText();
        System.out.println("Slate button text: " + slateText);

        driver.quit();
    }
}

File: Selenium/Activity4.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.color import Color

# Start browser
with webdriver.Firefox() as driver:

    # Open page
    driver.get("https://training-support.net/webelements/target-practice")

    # Print page title
    print("Page title:", driver.title)

    third_header = driver.find_element(By.XPATH, "//h3[contains(text(),'#3')]")
    print("Third header:", third_header.text)

    color_value = driver.find_element(
        By.XPATH, "//h5[contains(text(),'#5')]"
    ).value_of_css_property("color")

    color = Color.from_string(color_value)
    print("RGB:", color.rgb)
    print("HEX:", color.hex)

    purple_button = driver.find_element(By.XPATH, "//button[text()='Purple']")
    print("Purple classes:", purple_button.get_attribute("class"))

    slate_button = driver.find_element(By.XPATH, "//button[text()='Slate']")
    print("Slate button text:", slate_button.text)